### PR TITLE
test: Add bddtest structure

### LINKED ISSUE #50: Add bddtest structure

### PR DESCRIPTION
**Pull request by fqutishat**
closes #50

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Comment by troyronda** [comment]
The generated configs / fixtures should not be checked-in. In fabric-go-sdk, we have scripts to generate as part of the test process.

File: test/bddtests/fixtures/config/couchDB/config.ini [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#  Copyright SecureKey Technologies Inc.
```
Comment (troyronda) [inline test/bddtests/fixtures/config/couchDB/config.ini]: Wrong license header

File: test/bddtests/fixtures/config/mutual_tls/ca/client_sdk_go-ca_root.pem [hunk 2]
```diff
@@ -0,0 +1,18 @@
+-----BEGIN CERTIFICATE-----
```
Comment (troyronda) [inline test/bddtests/fixtures/config/mutual_tls/ca/client_sdk_go-ca_root.pem]: should also use generated fixtures. these should not be checked in.

File: test/bddtests/fixtures/config/prometheus/prometheus.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+   # Copyright SecureKey Technologies Inc.
```
Comment (troyronda) [inline test/bddtests/fixtures/config/prometheus/prometheus.yml]: wrong header

File: test/bddtests/fixtures/testdata/src/github.com/trustbloc/example_cc/base58.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2013-2015 The btcsuite developers
```
Comment (troyronda) [inline test/bddtests/fixtures/testdata/src/github.com/trustbloc/example_cc/base58.go]: Need an Apachev2 licensed version.

File: test/bddtests/fixtures/config/prometheus/prometheus.yml (hunk 3 above)
Comment (fqutishat) [reply test/bddtests/fixtures/config/prometheus/prometheus.yml]: done

File: test/bddtests/fixtures/testdata/src/github.com/trustbloc/example_cc/base58.go (hunk 4 above)
Comment (fqutishat) [reply test/bddtests/fixtures/testdata/src/github.com/trustbloc/example_cc/base58.go]: done

File: test/bddtests/fixtures/config/couchDB/config.ini (hunk 1 above)
Comment (fqutishat) [reply test/bddtests/fixtures/config/couchDB/config.ini]: done

File: test/bddtests/fixtures/config/mutual_tls/ca/client_sdk_go-ca_root.pem (hunk 2 above)
Comment (fqutishat) [reply test/bddtests/fixtures/config/mutual_tls/ca/client_sdk_go-ca_root.pem]: this is generated by open ssl

Comment (troyronda) [reply test/bddtests/fixtures/config/mutual_tls/ca/client_sdk_go-ca_root.pem]: take a look at the "tls.example.com" organization in the fabric-sdk-go.

Comment (fqutishat) [reply test/bddtests/fixtures/config/mutual_tls/ca/client_sdk_go-ca_root.pem]: done

**Comment by fqutishat** [comment]
verify-unitest

File: test/bddtests/fixtures/config/couchDB/config.ini [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#
+# Copyright IBM Corp, SecureKey Technologies Inc. All Rights Reserved.
```
Comment (troyronda) [inline test/bddtests/fixtures/config/couchDB/config.ini]: There were really two authors of a two line file?

Comment (fqutishat) [reply test/bddtests/fixtures/config/couchDB/config.ini]: done

**Pull request merged**